### PR TITLE
Marshmallow upgrade from 2.x to 3.x

### PR DESCRIPTION
**Pull request by alacasse**
J'ai suivi ce guide: https://marshmallow.readthedocs.io/en/stable/upgrading.html

### CODE REVIEW

File: tests/test_validation.py [hunk 1]
```diff
@@ -49,15 +49,16 @@ def __init__(self, *,
...
     assert data == {"target": "http://example.com/"}
 
 
 def test_custom_field_fails_validation():
-    a = A(target="example.com/")
-
-    data, errors = a.dump(strict=False)
-    assert errors == {"target": ["Not a valid URL."]}
+    try:
+        a = A(target="example.com/")
```
Comment (lphuberdeau) [inline tests/test_validation.py:58]: Test won't fail if there is no validation failure.

File: tests/test_validation.py [hunk 2]
```diff
@@ -75,5 +76,5 @@ def __init__(self, *, popularity: int, links: List[str]):
 def test_load_validation():
     MyPageRank(popularity=5, links=['http://www.foo.com']).dump()
     tampered_record = {'popularity': 1, 'links': ['1337']}
-    with raises(ValidationError):
-        MyPageRank.load(tampered_record)
+    with raises(TypeError):
```
Comment (lphuberdeau) [inline tests/test_validation.py:79]: This does not seem right. It should still be a ValidationError.

File: requirements.txt [hunk 3]
```diff
@@ -1,2 +1,2 @@
-marshmallow>=2.21,<3.0
+marshmallow>=3.0
```
Comment (lphuberdeau) [inline requirements.txt:1]: Should likely preserve the upper bound to <4.0 to prevent future breaking changes.

**Review by lphuberdeau — CHANGES_REQUESTED** [review]

File: tests/test_validation.py (hunk 1 above)
Comment (alacasse) [reply tests/test_validation.py:58]: I'm not sure what you mean here.

Comment (lphuberdeau) [reply tests/test_validation.py:58]: If no exception is thrown (which I believe is the case), there is no assert statement being executed.

**Comment by lphuberdeau** [comment]
Equivalent PR merged.

**Pull request closed without merging**